.github/workflows/ci-cd.yml

```
@@ -0,0 +1,165 @@
+      
+    - name: Run type checking
+      run: npm run type-check
+      
+    - name: Run unit tests
+      run: npm run test:unit
+      
+    - name: Run integration tests
+      run: npm run test:integration
+      
+    - name: Upload coverage reports to Codecov
+      uses: codecov/codecov-action@v3
```

actionlint (1.7.8)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)